src/Http/Middleware/IpHandler.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Http\Middleware;
+
+use Illuminate\Support\Arr;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface as Middleware;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class IpHandler implements Middleware
+{
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        $ipAddress = Arr::get($request->getServerParams(), 'REMOTE_ADDR', '127.0.0.1');
```

Comment:
Won't this break with cloudflare? They use a different header to pass the real IP

Reply:
The point of this PR is to centralize the logic for getting the PR in a Middleware, and set it as a request attribute. That way we don't have duplicate code for calculating it in multiple controllers. In the future this will be augmented to deal with reverse proxies and other stuff but for now it's just a refactor of where the logic is to prevent future BC breaks.

Reply:
Sounds good